code/LeftAndMain.php

```diff
@@ -874,17 +852,14 @@ public function redirect($url, $code = 302)
             }
             $newResponse->setIsFinished(true);
             $this->setResponse($newResponse);
-            return ''; // Actual response will be re-requested by client
+            // Actual response will be re-requested by client
+            return $newResponse;
         } else {
             return parent::redirect($url, $code);
         }
     }
 
-    /**
-     * @param HTTPRequest $request
-     * @return HTTPResponse
-     */
-    public function index($request)
+    public function index(HTTPRequest $request): HTTPResponse
```

Comment:
This will mean any custom `ModelAdmin` or other `LeftAndMain` subclasses that want to override `index()` will have to return `HTTPResponse` explicitly, where previously other types were possible. It's unclear if that fails the acceptance criteria since technically that's not a custom action....

Reply:
I think this is an OK change to make
a) It's a very obvious error to fix on upgrade to CMS 5 - you'll see it as soon as you go to /admin
b) It's unlikely that ModelAdmin's will even implement `index()` - none of the core classes that `extends ModelAdmin` implement `index()`

Reply:
Have retained and updated developer-docs